str/str_iterative_hipstr_runner.py

```diff
@@ -0,0 +1,137 @@
+# pylint: disable=import-error
+
+"""
+This script uses HipSTR to call STRs on WGS cram files, using the joint calling option. 
+For example:
+analysis-runner --access-level test --dataset hgdp --description 'hipstr run' --output-dir 'str/sensitivity-analysis/hipstr' str_iterative_hipstr_runner.py --variant-catalog=gs://cpg-hgdp-test/str/untrimmed_coordinates_resources/hg38_hipstr_catalog_untrimmed_coordinates.bed --output-file-name=hipster_90_genomes --dataset=hgdp HGDP00511
+
+Required packages: sample-metadata, hail, click, os
+pip install sample-metadata hail click
+
+"""
+import logging
+
+import os
+import click
+from sample_metadata.apis import AnalysisApi, SampleApi
+from sample_metadata.model.analysis_type import AnalysisType
+from sample_metadata.model.analysis_query_model import AnalysisQueryModel
+from sample_metadata.models import AnalysisStatus
+from cpg_utils.config import get_config
+from cpg_utils.hail_batch import output_path
+from cpg_workflows.batch import get_batch
+
+config = get_config()
+
+SAMTOOLS_IMAGE = config['images']['samtools']
+HIPSTR_IMAGE = config['images']['hipstr']
+BCFTOOLS_IMAGE = config['images']['bcftools']
+
+
+# inputs:
+@click.option('--variant-catalog', help='Full path to HipSTR Variants catalog')
+@click.option('--dataset', help='dataset eg tob-wgs')
+@click.argument('external-wgs-ids', nargs=-1)
+@click.option('--output-file-name', help='Output file name without file extension')
+@click.command()
+def get_cloudfuse_paths(dataset, external_wgs_ids):
+    """Retrieves cloud fuse paths and outputs as a comma-separated string for crams associated with the external-wgs-ids"""
+    # Extract CPG IDs from external sample IDs
+    external_id_to_cpg_id: dict[str, str] = SampleApi().get_sample_id_map_by_external(
+        dataset, list(external_wgs_ids)
+    )
+    analysis_query_model = AnalysisQueryModel(
+        sample_ids=list(external_id_to_cpg_id.values()),
+        projects=[dataset],
+        type=AnalysisType('cram'),
+        status=AnalysisStatus('completed'),
+        meta={'sequence_type': 'genome', 'source': 'nagim'},
+    )
+    crams_path = AnalysisApi().query_analyses(analysis_query_model)
+    cpg_sids_with_crams = set(sid for sids in crams_path for sid in sids['sample_ids'])
+    cpg_id_to_external_id = {
+        cpg_id: external_wgs_id
+        for external_wgs_id, cpg_id in external_id_to_cpg_id.items()
+    }
+    cpg_sids_without_crams = set(cpg_id_to_external_id.keys()) - cpg_sids_with_crams
+    if cpg_sids_without_crams:
+        external_wgs_sids_without_crams = ', '.join(
+            cpg_id_to_external_id[sid] for sid in cpg_sids_without_crams
+        )
+        logging.warning(
+            f'There were some samples without CRAMs: {external_wgs_sids_without_crams}'
+        )
+    # Create string containing paths based on /cramfuse
+    cramfuse_path = ''
+    for cram_obj in crams_path:
+        cpg_sample_id = os.path.basename(cram_obj['output']).split('.')[0]
```

Comment:
Can you use the path instead of trying to extract the filename. 
Because you're mounting the main bucket directly, then you could just strip the bucket name and use the path, eg:

```python
# removes the gs:// then, grab everything after the bucket name
suffix = cram_obj['output'].removeprefix('gs://').split('/', maxsplit=1)[1]
cramfuse_path = f'/cramfuse/{suffix}'
```